### ISSUE TITLE
Numerous issues with a fresh create-react-app install

### ISSUE THREAD

**Issue by farcaller** [welldone-software/why-did-you-render#149]
Using create-react-app 3.4.1 (which results in react 16.13.1)

## Setting up

```
$ create-react-app --typescript testapp
...
$ cd testapp
$ yarn add --dev @welldone-software/why-did-you-render
```

I updated `index.tsx` to read:

```tsx
import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import App from './App';
import * as serviceWorker from './serviceWorker';

//////////////// ADDED THIS
if (process.env.NODE_ENV === 'development') {
  const whyDidYouRender = require('@welldone-software/why-did-you-render');
  whyDidYouRender(React, {});
}
//////////////// UP TO THIS

ReactDOM.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  document.getElementById('root')
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: https://bit.ly/CRA-PWA
serviceWorker.unregister();
```

I updated `App.tsx` to read:

```tsx
import React, { useState } from 'react';

function App() {
  const [whatever, setWhatever] = useState(1);

  const clickme = () => {
    setWhatever(whatever + 1);
  }

  return (
    <div>
      <button onClick={clickme}>test</button>
      <div>{whatever}</div>
    </div>
  );
}

export default App;
```

## Expectations

1. I expect WDYR to work if I add `trackHooks: true` and either of:

1.1. Add `App.whyDidYouRender = true` before the `export default App` line
1.2. Add `trackAllPureComponents: true` and change `export default App` to read `export default React.memo(App)`

## What actually happens

Both 1.1 and 1.2 produce no logs

**Comment by vzaidman** [comment]
try

```
function App() {
  const [whatever, setWhatever] = useState({});

  const clickme = () => {
    setWhatever({});
  }

  return (
    <div>
      <button onClick={clickme}>test</button>
      <div>{whatever}</div>
    </div>
  );
}
```

**Comment by vzaidman** [comment]
also what WDYR version do you use?

**Comment by vzaidman** [comment]
also see the installation guides
https://github.com/welldone-software/why-did-you-render#installation

**Comment by vzaidman** [comment]
also, it seems to work in a similar sandbox:
https://codesandbox.io/s/cocky-jepsen-mjtml?file=/src/index.js

take a look please if there's any difference to what you did

**Comment by farcaller** [comment]
It seems npm pulled in pre-released 5.0.0-alpha.2. It works all fine on 4.3.1.

**Comment by vzaidman** [comment]
thanks for the report. will be fixed in beta. i know what's the issue there. meanwhile please use 4.3.1

**Comment by vzaidman** [comment]
fixed in [` 5.0.0-alpha.4`](https://github.com/welldone-software/why-did-you-render/releases/tag/v5.0.0-alpha.4)
